docs/docs/databases/snowflake.mdx

```diff
@@ -29,3 +29,38 @@ user/role rights during engine creation by default. However, when pressing the 
 button in the Create or Edit Database dialog, user/role credentials are validated by passing
 “validate_default_parameters”: True to the connect() method during engine creation. If the user/role
 is not authorized to access the database, an error is recorded in the Superset logs.
+
+And if you want connect Snowflake with [Key Pair Authentication](https://docs.snowflake.com/en/user-guide/key-pair-auth.html#step-6-configure-the-snowflake-client-to-use-key-pair-authentication).
+Plase make sure you have the key pair and the public key is registered in Snowflake.
+To connect Snowflake with Key Pair Authentication, you need to add the following parameters to "SECURE EXTRA" field.
+
+***Please note that you need to split your private key content line by line before store it to privatekey_body*** 
+
+```
+{
+    "auth_method": "keypair", 
+    "auth_params": {
+        "privatekey_body":[
+            "-----BEGIN ENCRYPTED PRIVATE KEY-----",
+            "Key Body",
+            "Key Body",
+            "Key Body",
+            "Key Body",
+            "-----END ENCRYPTED PRIVATE KEY-----"
+        ],
+        "privatekey_pass":"Your Private Key Password"
+    }
+}
+```
```

Comment:
While JSON isn't ideal for storing multiline strings, maybe we should just store it with the newlines replaced by a `\n` as follows:
```json
{
     "auth_method": "keypair", 
     "auth_params": {
         "privatekey_body": "-----BEGIN ENCRYPTED PRIVATE KEY-----\n...\n-----END ENCRYPTED PRIVATE KEY-----",
         "privatekey_pass":"Your Private Key Password"
     }
 }
```
Later when we consolidate this type of logic across all db engine specs we can then move the private key to a separate column in the database model and have a dedicated field in the UI that accepts the string with newlines. WDYT @xiayanzheng @dpgaspar ?

Reply:
@villebro a string with newlines would be more user friendly, JSON is more flexible but can tend to generate more divergent configs. standardization pass for 3.0 sounds perfect to me.

Currently, I prefer the `\n` way for `privatekey_body`

Reply:
Thanks for your advice, I'm ok with both formats, and this change just pushed to this PR.